TeleportMenu.lua

```diff
@@ -667,29 +647,50 @@ end
 function tpm:updateAvailableSeasonalTeleport()
 	local playerFaction = UnitFactionGroup("player")
 	local siegeOfBoralus = -1
+	local motherlode = -1
 	if playerFaction == "Alliance" then
 		siegeOfBoralus = 445418
+		motherlode = 467553
 	else
 		siegeOfBoralus = 464256
+		motherlode = 467555
 	end
 
-	local challengeMapIdTospellID = {
-		[353] = siegeOfBoralus, -- Siege of Boralus has two spells one for alliance and one for horde
-		[375] = 354464, -- Mists
-		[376] = 354462, -- Necrotic Wake
-		[499] = 445444, -- Priory
-		[500] = 445443, -- The Rookery
-		[501] = 445269, -- Stonevault
-		[502] = 445416, -- City of Threads
-		[503] = 445417, -- Ara Ara
-		[504] = 445441, -- Darkflame Cleft
-		[505] = 445414, -- The Dawnbreaker
-		[506] = 445440, -- Cinderbrew Meadery
-		[507] = 445424 -- Grim Batol
+	local seasonalTeleports = {
+		[LE_EXPANSION_WAR_WITHIN] = { -- The War Within
+			[1] = { -- Season 1
+				[353] = siegeOfBoralus, -- Siege of Boralus has two spells one for alliance and one for horde
+				[375] = 354464, -- Mists
+				[376] = 354462, -- Necrotic Wake
+				[501] = 445269, -- Stonevault
+				[502] = 445416, -- City of Threads
+				[503] = 445417, -- Ara Ara
+				[505] = 445414, -- The Dawnbreaker
+				[507] = 445424, -- Grim Batol
+			},
+			[2] = { -- Season 2
+				[247] = motherlode, -- The MOTHERLODE!!
+				[370] = 373274, -- Operation: Mechagon - Workshop
+				[382] = 354467, -- Theater of Pain
+				[499] = 445444, -- Priory of the Sacred Flame
+				[500] = 445443, -- The Rookery
+				[504] = 445441, -- Darkflame Cleft
+				[506] = 445440, -- Cinderbrew Meadery
+				[525] = 1216786, -- Operation: Floodgate
+			},
+		},
 	}
 
+	local currentExpansion = GetExpansionLevel()
+	local currentSeason = C_MythicPlus.GetCurrentUIDisplaySeason()
+	if not currentSeason or currentSeason == 0 then
+		return
```

Comment:
Here we can set it to the last season hardcoded as fallback, 

`currentSeason = 1 -- Previous season as fallback incase we're inbetween seasons`